File: 1009.c
#define _CRT_SECURE_NO_WARNINGS

#include <stdio.h>
#include <math.h>

int main()
{
	int t;
	scanf("%d", &t);
	while (t--)
	{
		int garbage = 1;
		int a, b, init, cnt = 1, fin, temp;
		scanf("%d %d", &a, &b);
		init = a % 10;
		fin = a * a % 10;
		while (init != fin)
		{
			fin = fin * a % 10;
			cnt++; //주기
		}
		temp = (b - 1) % cnt + 1; //주기 중 몇번째?
		printf("%d\n", ((int)pow(a, temp) - 1) % 10 + 1);
	}
}

File: python/1009.py
from sys import stdin
t = int(stdin.readline())
for i in range(t):
    a,b = map(int, stdin.readline().rsplit())
    cnt = 1
    if a==1:
        print(1)
        continue
    init = a % 10
    fin = (a * a) % 10
    while init != fin:
        fin = (fin * a) % 10
        cnt += 1
    temp = (b - 1) % cnt + 1
    print((((a**temp)-1)%10)+1)
